crates/orchestrator/src/jobs/da_job/mod.rs

```diff
@@ -198,30 +203,28 @@ pub fn convert_to_biguint(elements: Vec<FieldElement>) -> Vec<BigUint> {
     biguint_vec
 }
 
-fn data_to_blobs(blob_size: u64, block_data: Vec<BigUint>) -> Result<Vec<Vec<u8>>, JobError> {
+pub fn data_to_blobs(blob_size: u64, block_data: Vec<BigUint>) -> Result<Vec<Vec<u8>>, JobError> {
     // Validate blob size
     if blob_size < 32 {
         Err(DaError::InsufficientBlobSize { blob_size })?
     }
 
     let mut blobs: Vec<Vec<u8>> = Vec::new();
 
-    // Convert all FieldElements to bytes first
-    let mut bytes: Vec<u8> = block_data.iter().flat_map(|element| element.to_bytes_be().to_vec()).collect();
+    // Convert all BigUint to bytes more efficiently
+    let bytes: Vec<u8> = block_data.iter().flat_map(BigUint::to_bytes_be).collect();
 
     // Process bytes in chunks of blob_size
-    while bytes.len() >= blob_size as usize {
-        let chunk = bytes.drain(..blob_size as usize).collect();
-        blobs.push(chunk);
-    }
-
-    // Handle any remaining bytes (not a complete blob)
-    if !bytes.is_empty() {
-        let remaining_bytes = bytes.len();
-        let mut last_blob = bytes;
-        last_blob.resize(blob_size as usize, 0); // Pad with zeros
-        blobs.push(last_blob);
-        log::debug!("Warning: Remaining {} bytes not forming a complete blob were padded", remaining_bytes);
+    let chunk_size = blob_size as usize;
+    let chunks = bytes.chunks(chunk_size);
+
+    for chunk in chunks {
+        let mut blob = chunk.to_vec();
+        if blob.len() < chunk_size {
+            blob.resize(chunk_size, 0);
+            log::debug!("Warning: Last chunk of {} bytes was padded to full blob size", chunk.len());
```

Comment:
should we throw an error if it's not the last?

Reply:
Idts we should throw an error in this case, as it is not an error as blob may have length less than the blob size.